### PR TITLE
test: e2e for auto-lock 

### PR DESCRIPTION
**Pull request by plasmacorral**
## **Description**

This PR adds a regression test to verify that changing the auto-lock setting to 'Immediately' requires a password when bringing the app from foreground to background. The test first backgrounds the app with the default auto-lock setting of 30 seconds, confirming that no password is required to access the wallet view. Then, it updates the auto-lock setting to 'Immediately' and verifies that a password is required to access the wallet view after re-launching the app.

## **Related issues**

Fixes: [551](https://github.com/MetaMask/accounts-planning/issues/551)

## **Manual testing steps**

Manual testing steps
On local machine run on this branch...
Android: `yarn test:e2e:android:debug:build` then `yarn test:e2e:android:debug:run ./e2e/specs/accounts/auto-lock.spec.js`
iOS: `yarn test:e2e:ios:debug:build` then `yarn test:e2e:ios:debug:run ./e2e/specs/accounts/auto-lock.spec.js`

pr_regression_e2e_pipeline: [Passed](https://app.bitrise.io/app/be69d4368ee7e86d/pipelines/cb74bbd7-d552-445e-adab-ecd727ba4b58)

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by plasmacorral** [comment]
smoke tests seem to be failing across branches with similar errors.  Thread [here](https://consensys.slack.com/archives/C02U025CVU4/p1724840900113059).

**Comment by plasmacorral** [comment]
New tests are passing in regression run, and known smoke flake is under investigation.

File: e2e/pages/Settings/SecurityAndPrivacy/SecurityAndPrivacyView.js [hunk 1]
```diff
@@ -42,6 +42,17 @@ class SecurityAndPrivacy {
...
   }
 
+  get autoLockSection() {
+    return Matchers.getElementByID(
+      SecurityPrivacyViewSelectorsIDs.AUTO_LOCK_SECTION,
+    );
+  }
+
+  get autoLockDefault30Seconds() {
+    const AUTO_LOCK_DEFAULT = 'After 30 seconds';
```
Comment (SamuelSalas) [inline e2e/pages/Settings/SecurityAndPrivacy/SecurityAndPrivacyView.js:52]: Ideally, all text selectors should be saved in the `SecurityPrivacyViewSelectorsText ` inside `e2e/selectors/Settings/SecurityAndPrivacy/SecurityPrivacyView.selectors`

Comment (plasmacorral) [reply e2e/pages/Settings/SecurityAndPrivacy/SecurityAndPrivacyView.js]: Good suggestion, thank you!  Done in 7b407580f046ca158964786e5d8f7199e1994493

Comment (plasmacorral) [reply e2e/pages/Settings/SecurityAndPrivacy/SecurityAndPrivacyView.js]: With this change updated regression run passes

**Review by SamuelSalas — APPROVED** [review]
LGTM

**Review by owencraston — APPROVED** [review]

**Pull request merged**